Chrysalis-Papilio/Soen490-Papilio-Website
# Use form data hook

## General Information

**Issue number**: Papilio-65

**Task description:**
Expand useFormData hook to implement input validation
- add register method to add input to the form 
- validate unit test of form related components
- add validation on submit and on blur

## Manual Testing
To test the validation follow these steps:
1. Make sure that you have the correct variables in the .env file 
2. start the backend
```npm run build && npm start```
3. start the website
```npm start```
4. On the first page, click the submit button. You should see the error message. 
5. Click the input then add a few letters then remove everything. 
6. Then add some information in the input then click submit. You should now be redirected to next page.

## Testing 
To run all unit tests, run the following code: ```npm test```
To run the coverage, run the following code: ```npm run coverage```
